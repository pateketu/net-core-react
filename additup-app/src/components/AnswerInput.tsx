import * as React from 'react';
import IAnswerProps from './IAnswerProps';

export const AnswerInput: React.SFC<IAnswerProps> = (props) => {
    return <div>
                <div className="equation">
                    <div>{props.a}</div>
                    <div>+</div>
                    <div>{props.b}</div>
                    <div>=</div>
                    <div><input type="text"
                            ref={(input) => input && input.focus() }
                            onChange={(input) =>
                             props.onAnswer(
                                input.target.value.trim().length > 0 ?
                                    Number(input.target.value) :
                                    NaN)
                        }/></div>
                </div>
            </div>;
};
