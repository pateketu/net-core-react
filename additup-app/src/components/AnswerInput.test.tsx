import * as React from 'react';
import renderer from 'react-test-renderer';
import { AnswerInput} from './AnswerInput';
import {shallow} from 'enzyme';

describe('Answer Input Component', () => {
    it('Renders expected html mark-up', () => {
        // tslint:disable-next-line:no-empty
        const tree = renderer.create(<AnswerInput a={1} b={2} onAnswer={() => {}}></AnswerInput>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Fires onAnswer with expected answer value', () => {
        let result: number = 0;
        const value: number = 20;

        const wrapper = shallow(<AnswerInput a={1} b={2}
            onAnswer={(answer: number ) => { result = answer; }}></AnswerInput>);

        wrapper.find('input').simulate('change', {target: {value: String(value)}});

        expect(result).toBe(value);
    });

    it('Fires onAnswer with NaN value', () => {
        let result: number = 0;

        const wrapper = shallow(<AnswerInput a={1} b={2}
            onAnswer={(answer: number ) => { result = answer; }}></AnswerInput>);

        wrapper.find('input').simulate('change', {target: {value: 'foo'}});

        expect(result).toBe(NaN);
    });
});
