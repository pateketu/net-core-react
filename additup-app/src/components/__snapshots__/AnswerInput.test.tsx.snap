// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Input Component Renders expected html mark-up 1`] = `
<div>
  <div
    className="equation"
  >
    <div>
      1
    </div>
    <div>
      +
    </div>
    <div>
      2
    </div>
    <div>
      =
    </div>
    <div>
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
</div>
`;
